testing/path.go

```diff
@@ -83,3 +85,54 @@ func (path *Path) RelayPacket(packet channeltypes.Packet) error {
 
 	return fmt.Errorf("packet commitment does not exist on either endpoint for provided packet")
 }
+
+// RelayPacketWithResults attempts to relay the packet first on EndpointA and then on EndpointB
+// if EndpointA does not contain a packet commitment for that packet. The function returns:
+// - The result of the packet receive transaction.
+// - The acknowledgement written on the receiving chain.
+// - An error if a relay step fails or the packet commitment does not exist on either endpoint.
+func (path *Path) RelayPacketWithResults(packet channeltypes.Packet) (res *sdk.Result, ack []byte, err error) {
+	pc := path.EndpointA.Chain.App.GetIBCKeeper().ChannelKeeper.GetPacketCommitment(path.EndpointA.Chain.GetContext(), packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
```

Comment:
to reduce duplication, we can now write `RelayPacket` as a wrapper of this which just drops the res, ack returns no?

Reply:
yes, great idea!